[CI] DLMFrozenTransitionDisruptionIT testConcurrentRolloverDuringTransition failing

**Build Scans:**
- [elasticsearch-intake #39645 / part2](https://gradle-enterprise.elastic.co/s/eexp5i4cncnh2)
- [elasticsearch-pull-request #144814 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/44y65ghs6zqrq)
- [elasticsearch-pull-request #144814 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/dtb47ipnnpuko)
- [elasticsearch-periodic-java-ea #267 / release-tests](https://gradle-enterprise.elastic.co/s/ntasawnvzk2aq)
- [elasticsearch-pull-request #144754 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/ahn5gnxgrhciu)
- [elasticsearch-pull-request #144754 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/pckmvw5ifcsri)
- [elasticsearch-pull-request #144750 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/pgezx5hfc3dlm)
- [elasticsearch-pull-request #144750 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/ttmpygcqrvh6q)
- [elasticsearch-pull-request #144733 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/dzbtba7fjdxv2)
- [elasticsearch-pull-request #144733 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/potw4quqryrm6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:dlm-frozen-transition:internalClusterTest" --tests "org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionDisruptionIT.testConcurrentRolloverDuringTransition" -Dtests.seed=DDC37C3589ADB347 -Dtests.locale=kln-Latn-KE -Dtests.timezone=Asia/Thimbu -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionDisruptionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testConcurrentRolloverDuringTransition'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was completed exceptionally
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testConcurrentRolloverDuringTransition
- [main] 1 consecutive failures in step part2
- [main] 30 consecutive failures in step checkpart2_release-tests
- [main] 1 consecutive failures in step release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 32 failures in test testConcurrentRolloverDuringTransition (4.7% fail rate in 682 executions)
- [main] 30 failures in step checkpart2_release-tests (100.0% fail rate in 30 executions)
- [main] 15 failures in pipeline elasticsearch-pull-request (3.2% fail rate in 466 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -480,9 +480,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.snapshots.StatelessSnapshotIT
   method: testSnapshotFailsCleanlyWhenShardClosesDuringDisconnect
   issue: https://github.com/elastic/elasticsearch/issues/149032
-- class: org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionDisruptionIT
-  method: testConcurrentRolloverDuringTransition
-  issue: https://github.com/elastic/elasticsearch/issues/149039
 - class: org.elasticsearch.packaging.test.BootstrapCheckTests
   method: test20RunWithBootstrapChecks
   issue: https://github.com/elastic/elasticsearch/issues/149040

--- a/x-pack/plugin/dlm-frozen-transition/src/internalClusterTest/java/org/elasticsearch/xpack/dlm/frozen/DLMFrozenTransitionDisruptionIT.java
+++ b/x-pack/plugin/dlm-frozen-transition/src/internalClusterTest/java/org/elasticsearch/xpack/dlm/frozen/DLMFrozenTransitionDisruptionIT.java
@@ -423,9 +423,14 @@ public class DLMFrozenTransitionDisruptionIT extends ESIntegTestCase {
         );
         triggerRollover();
 
+        // The second backing index (created by triggerRollover) will also become a candidate
+        // after the concurrent rollover creates a third backing index.
+        String secondBackingIndex = backingIndexNames().get(1);
+
         assertTrue("ForceMerge request was never seen by the interceptor", latch.await(30, TimeUnit.SECONDS));
 
         String expectedFrozenIndexName = DLMConvertToFrozen.SNAPSHOT_NAME_PREFIX + candidateIndex;
+        String expectedFrozenIndexName2 = DLMConvertToFrozen.SNAPSHOT_NAME_PREFIX + secondBackingIndex;
         assertBusy(() -> {
             var projectMetadata = clusterAdmin().prepareState(TEST_REQUEST_TIMEOUT)
                 .get()
@@ -433,13 +438,18 @@ public class DLMFrozenTransitionDisruptionIT extends ESIntegTestCase {
                 .metadata()
                 .getProject(Metadata.DEFAULT_PROJECT_ID);
             assertThat(
-                "Frozen index should exist despite concurrent rollover",
+                "Frozen index should exist for first candidate despite concurrent rollover",
                 projectMetadata.index(expectedFrozenIndexName),
+                notNullValue()
+            );
+            assertThat(
+                "Frozen index should exist for second candidate after concurrent rollover",
+                projectMetadata.index(expectedFrozenIndexName2),
                 notNullValue()
             );
         }, 120, TimeUnit.SECONDS);
 
-        logger.info("--> concurrent rollover during transition handled gracefully");
+        logger.info("--> concurrent rollover during transition handled gracefully, both indices converted to frozen");
     }
 
     /**

